inc/functions.php

```diff
@@ -1014,7 +1017,7 @@ function redirect($url, $message="", $title="", $force_redirect=false)
 	}
 
 	// Show redirects only if both ACP and UCP settings are enabled, or ACP is enabled, and user is a guest, or they are forced.
-	if($force_redirect == true || ($mybb->settings['redirects'] == 1 && ($mybb->user['showredirect'] == 1 || !$mybb->user['uid'])))
+	if($force_redirect == true || ($mybb->settings['redirects'] == 1 && (!$mybb->user['uid'] || !$mybb->user['showredirect'] == 1)))
```

Comment:
@dvz `$mybb->user['showredirect'] == 1` was accidentally changed to `!$mybb->user['showredirect'] == 1`. Note the exclamation mark.